diff -Npur plasma-desktop-5.20.5.orig/kcms/users/package/contents/ui/FingerprintDialog.qml plasma-desktop-5.20.5/kcms/users/package/contents/ui/FingerprintDialog.qml
--- plasma-desktop-5.20.5.orig/kcms/users/package/contents/ui/FingerprintDialog.qml	1970-01-01 01:00:00.000000000 +0100
+++ plasma-desktop-5.20.5/kcms/users/package/contents/ui/FingerprintDialog.qml	2021-01-23 14:04:18.477355122 +0100
@@ -0,0 +1,261 @@
+/*
+    Copyright 2020  Devin Lin <espidev@gmail.com>
+
+    This library is free software; you can redistribute it and/or
+    modify it under the terms of the GNU Lesser General Public
+    License as published by the Free Software Foundation; either
+    version 2.1 of the License, or (at your option) version 3, or any
+    later version accepted by the membership of KDE e.V. (or its
+    successor approved by the membership of KDE e.V.), which shall
+    act as a proxy defined in Section 6 of version 3 of the license.
+
+    This library is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+    Lesser General Public License for more details.
+
+    You should have received a copy of the GNU Lesser General Public
+    License along with this library.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import QtQuick 2.12
+import QtQuick.Dialogs 1.1
+import QtQuick.Layouts 1.3
+import QtQuick.Shapes 1.12
+import QtQuick.Controls 2.5 as QQC2
+
+import org.kde.kirigami 2.12 as Kirigami
+import FingerprintModel 1.0
+
+Kirigami.OverlaySheet {
+    id: fingerprintRoot
+    
+    property var account
+    property var fingerprintModel: kcm.fingerprintModel
+    property string currentFinger
+    
+    enum DialogState {
+        FingerprintList,
+        PickFinger,
+        Enrolling,
+        EnrollComplete
+    }
+    
+    function openAndClear() {
+        fingerprintModel.switchUser(account.name == kcm.userModel.getLoggedInUser().name ? "" : account.name);
+        this.open();
+    }
+
+    onSheetOpenChanged: {
+        if (sheetOpen && fingerprintModel.currentlyEnrolling) {
+            fingerprintModel.stopEnrolling();
+        }
+    }
+    
+    header: Kirigami.Heading {
+        text: i18n("Fingerprint")
+    }
+    
+    footer: RowLayout {
+        Item {
+            Layout.fillWidth: true
+        }
+        
+        // FingerprintList State
+        QQC2.Button {
+            text: i18n("Clear Fingerprints")
+            visible: fingerprintModel.dialogState === FingerprintDialog.DialogState.FingerprintList
+            enabled: fingerprintModel.enrolledFingerprints.length !== 0
+            icon.name: "delete"
+            onClicked: fingerprintModel.clearFingerprints();
+        }
+        QQC2.Button {
+            text: i18n("Add")
+            visible: fingerprintModel.dialogState === FingerprintDialog.DialogState.FingerprintList
+            enabled: fingerprintModel.availableFingersToEnroll.length !== 0
+            icon.name: "list-add"
+            onClicked: fingerprintModel.dialogState = FingerprintDialog.DialogState.PickFinger
+        }
+        
+        // PickFinger State
+        QQC2.Button {
+            text: i18n("Cancel")
+            visible: fingerprintModel.dialogState === FingerprintDialog.DialogState.PickFinger
+            icon.name: "dialog-cancel"
+            onClicked: fingerprintModel.dialogState = FingerprintDialog.DialogState.FingerprintList
+        }
+        QQC2.Button {
+            text: i18n("Continue")
+            visible: fingerprintModel.dialogState === FingerprintDialog.DialogState.PickFinger
+            icon.name: "dialog-ok"
+            onClicked: {
+                fingerprintRoot.currentFinger = pickFingerBox.currentText
+                fingerprintModel.startEnrolling(pickFingerBox.currentValue);
+            }
+        }
+        
+        // Enrolling State
+        QQC2.Button {
+            text: i18n("Cancel")
+            visible: fingerprintModel.dialogState === FingerprintDialog.DialogState.Enrolling
+            icon.name: "dialog-cancel"
+            onClicked: fingerprintModel.stopEnrolling();
+        }
+        
+        // EnrollComplete State
+        QQC2.Button {
+            text: i18n("Done")
+            visible: fingerprintModel.dialogState === FingerprintDialog.DialogState.EnrollComplete
+            icon.name: "dialog-ok"
+            onClicked: fingerprintModel.stopEnrolling();
+        }
+    }
+
+    contentItem: Item {
+        id: rootPanel
+        // TODO figure out why specified width is not being used at all
+        Layout.preferredWidth: Kirigami.Units.gridUnit * 12
+        Layout.maximumWidth: Kirigami.Units.gridUnit * 12
+        Layout.leftMargin: Kirigami.Units.smallSpacing
+        Layout.rightMargin: Kirigami.Units.smallSpacing
+        width: Kirigami.Units.gridUnit * 12
+        height: Kirigami.Units.gridUnit * 12
+        
+        ColumnLayout {
+            id: enrollFeedback
+            spacing: Kirigami.Units.largeSpacing * 2
+            visible: fingerprintModel.dialogState === FingerprintDialog.DialogState.Enrolling || fingerprintModel.dialogState === FingerprintDialog.DialogState.EnrollComplete
+            anchors.fill: parent
+            
+            Kirigami.Heading {
+                level: 2
+                text: i18n("Enrolling Fingerprint")
+                Layout.alignment: Qt.AlignHCenter
+                visible: fingerprintModel.dialogState === FingerprintDialog.DialogState.Enrolling
+            }
+            
+            QQC2.Label {
+                text: i18n("Please repeatedly " + fingerprintModel.scanType + " your " + fingerprintRoot.currentFinger.toLowerCase() + " on the fingerprint sensor.")
+                Layout.alignment: Qt.AlignHCenter
+                wrapMode: Text.Wrap
+                horizontalAlignment: Text.AlignHCenter
+                Layout.maximumWidth: parent.width
+                visible: fingerprintModel.dialogState === FingerprintDialog.DialogState.Enrolling
+            }
+            
+            Kirigami.Heading {
+                level: 2
+                text: i18n("Finger Enrolled")
+                Layout.alignment: Qt.AlignHCenter
+                visible: fingerprintModel.dialogState === FingerprintDialog.DialogState.EnrollComplete
+            }
+
+            // reset from back from whatever color was used before
+            onVisibleChanged: progressCircle.colorTimer.restart();
+            
+            // progress circle
+            FingerprintProgressCircle {
+                id: progressCircle
+            }
+            
+            QQC2.Label {
+                text: fingerprintModel.enrollFeedback
+                wrapMode: Text.Wrap
+                Layout.maximumWidth: parent.width
+                Layout.alignment: Qt.AlignVCenter | Qt.AlignHCenter
+            }
+        }
+        
+        ColumnLayout {
+            id: pickFinger
+            visible: fingerprintModel.dialogState === FingerprintDialog.DialogState.PickFinger
+            anchors.centerIn: parent
+            spacing: Kirigami.Units.largeSpacing
+            
+            Kirigami.Icon {
+                source: "fingerprint"
+                implicitHeight: Kirigami.Units.iconSizes.huge
+                implicitWidth: Kirigami.Units.iconSizes.huge
+                Layout.alignment: Qt.AlignHCenter
+            }
+            
+            Kirigami.Heading {
+                level: 2
+                text: i18n("Pick a finger to enroll")
+                Layout.alignment: Qt.AlignHCenter
+            }
+            
+            QQC2.ComboBox {
+                id: pickFingerBox
+                model: fingerprintModel.availableFingersToEnroll
+                textRole: "friendlyName"
+                valueRole: "internalName"
+                Layout.alignment: Qt.AlignHCenter
+                onActivated: fingerprintRoot.currentFinger = currentText
+            }
+        }
+        
+        ColumnLayout {
+            id: fingerprints
+            spacing: Kirigami.Units.smallSpacing
+            visible: fingerprintModel.dialogState === FingerprintDialog.DialogState.FingerprintList
+            anchors.fill: parent
+            
+            Kirigami.InlineMessage {
+                id: errorMessage
+                type: Kirigami.MessageType.Error
+                visible: fingerprintModel.currentError !== ""
+                text: fingerprintModel.currentError
+                Layout.fillWidth: true
+                actions: [
+                    Kirigami.Action {
+                        iconName: "dialog-close"
+                        onTriggered: fingerprintModel.currentError = ""
+                    }
+                ]
+            }
+            
+            ListView {
+                id: fingerprintsList
+                model: kcm.fingerprintModel.deviceFound ? fingerprintModel.enrolledFingerprints : 0
+                Layout.fillWidth: true
+                Layout.fillHeight: true
+                QQC2.ScrollBar.vertical: QQC2.ScrollBar {}
+
+                delegate: Kirigami.SwipeListItem {
+                    property Finger finger: modelData
+                    contentItem: RowLayout {
+                        Kirigami.Icon {
+                            source: "fingerprint"
+                            height: Kirigami.Units.iconSizes.medium
+                            width: Kirigami.Units.iconSizes.medium
+                        }
+                        QQC2.Label {
+                            Layout.fillWidth: true
+                            text: finger.friendlyName
+                        }
+                    }
+                    actions: [
+                        Kirigami.Action {
+                            iconName: "edit-entry"
+                            onTriggered: {
+                                fingerprintRoot.currentFinger = finger.friendlyName;
+                                fingerprintModel.startEnrolling(finger.internalName);
+                            }
+                            tooltip: i18n("Re-enroll finger")
+                        }
+                    ]
+                }
+                
+                Kirigami.PlaceholderMessage {
+                    anchors.centerIn: parent
+                    width: parent.width - (Kirigami.Units.largeSpacing * 4)
+                    visible: fingerprintsList.count == 0
+                    text: i18n("No fingerprints added")
+                    icon.name: "fingerprint"
+                }
+            }
+        }
+    }
+}
+
diff -Npur plasma-desktop-5.20.5.orig/kcms/users/package/contents/ui/FingerprintProgressCircle.qml plasma-desktop-5.20.5/kcms/users/package/contents/ui/FingerprintProgressCircle.qml
--- plasma-desktop-5.20.5.orig/kcms/users/package/contents/ui/FingerprintProgressCircle.qml	1970-01-01 01:00:00.000000000 +0100
+++ plasma-desktop-5.20.5/kcms/users/package/contents/ui/FingerprintProgressCircle.qml	2021-01-23 14:04:18.477355122 +0100
@@ -0,0 +1,131 @@
+/*
+    Copyright 2020  Devin Lin <espidev@gmail.com>
+
+    This library is free software; you can redistribute it and/or
+    modify it under the terms of the GNU Lesser General Public
+    License as published by the Free Software Foundation; either
+    version 2.1 of the License, or (at your option) version 3, or any
+    later version accepted by the membership of KDE e.V. (or its
+    successor approved by the membership of KDE e.V.), which shall
+    act as a proxy defined in Section 6 of version 3 of the license.
+
+    This library is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+    Lesser General Public License for more details.
+
+    You should have received a copy of the GNU Lesser General Public
+    License along with this library.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import QtQuick 2.12
+import QtQuick.Layouts 1.3
+import QtQuick.Shapes 1.12
+
+import org.kde.kirigami 2.12 as Kirigami
+
+Item {
+    width: progressCircle.width
+    height: progressCircle.height
+    Layout.alignment: Qt.AlignVCenter | Qt.AlignHCenter
+    
+    property alias colorTimer: colorChangeBackTimer
+    
+    Timer {
+        id: colorChangeBackTimer
+        interval: 500
+        onTriggered: {
+            iconColorAnimation.to = Kirigami.Theme.textColor
+            iconColorAnimation.start();
+            circleColorAnimation.to = Kirigami.Theme.highlightColor
+            circleColorAnimation.start();
+        }
+    }
+    
+    Connections {
+        target: fingerprintModel
+        function onScanSuccess() {
+            iconColorAnimation.to = Kirigami.Theme.highlightColor
+            iconColorAnimation.start();
+            colorChangeBackTimer.restart();
+        }
+        function onScanFailure() {
+            iconColorAnimation.to = Kirigami.Theme.negativeTextColor
+            iconColorAnimation.start();
+            colorChangeBackTimer.restart();
+        }
+        function onScanComplete() {
+            iconColorAnimation.to = Kirigami.Theme.positiveTextColor
+            iconColorAnimation.start();
+        }
+    }
+    
+    Kirigami.Icon {
+        id: fingerprintEnrollFeedback
+        source: "fingerprint"
+        implicitHeight: Kirigami.Units.iconSizes.huge
+        implicitWidth: implicitHeight
+        anchors.centerIn: parent
+    
+        ColorAnimation on color {
+            id: iconColorAnimation
+            easing.type: Easing.InOutQuad
+            duration: 150
+        }
+    }
+    
+    Shape {
+        id: progressCircle
+        anchors.horizontalCenter: fingerprintEnrollFeedback.horizontalCenter
+        anchors.verticalCenter: fingerprintEnrollFeedback.verticalCenter
+        implicitWidth: Kirigami.Units.iconSizes.huge + Kirigami.Units.gridUnit
+        implicitHeight: Kirigami.Units.iconSizes.huge + Kirigami.Units.gridUnit
+        layer.enabled: true
+        layer.samples: 40
+        anchors.centerIn: parent
+        
+        property int rawAngle: fingerprintModel.enrollProgress * 360
+        property int renderedAngle: 0
+        NumberAnimation on renderedAngle {
+            id: elapsedAngleAnimation
+            easing.type: Easing.InOutQuad
+            duration: 500
+        }
+        onRawAngleChanged: {
+            elapsedAngleAnimation.to = rawAngle;
+            elapsedAngleAnimation.start();
+        }
+        
+        ShapePath {
+            strokeColor: "lightgrey"
+            fillColor: "transparent"
+            strokeWidth: 3
+            capStyle: ShapePath.FlatCap
+            PathAngleArc {
+                centerX: progressCircle.implicitWidth / 2; centerY: progressCircle.implicitHeight / 2;
+                radiusX: (progressCircle.implicitWidth - Kirigami.Units.gridUnit) / 2; radiusY: radiusX;
+                startAngle: 0
+                sweepAngle: 360
+            }
+        }
+        ShapePath {
+            strokeColor: Kirigami.Theme.highlightColor
+            fillColor: "transparent"
+            strokeWidth: 3
+            capStyle: ShapePath.RoundCap
+            
+            ColorAnimation on strokeColor {
+                id: circleColorAnimation
+                easing.type: Easing.InOutQuad
+                duration: 200
+            }
+            
+            PathAngleArc {
+                centerX: progressCircle.implicitWidth / 2; centerY: progressCircle.implicitHeight / 2;
+                radiusX: (progressCircle.implicitWidth - Kirigami.Units.gridUnit) / 2; radiusY: radiusX;
+                startAngle: -90
+                sweepAngle: progressCircle.renderedAngle
+            }
+        }
+    }
+}
diff -Npur plasma-desktop-5.20.5.orig/kcms/users/package/contents/ui/UserDetailsPage.qml plasma-desktop-5.20.5/kcms/users/package/contents/ui/UserDetailsPage.qml
--- plasma-desktop-5.20.5.orig/kcms/users/package/contents/ui/UserDetailsPage.qml	2021-01-05 13:13:03.000000000 +0100
+++ plasma-desktop-5.20.5/kcms/users/package/contents/ui/UserDetailsPage.qml	2021-01-23 14:08:53.184029285 +0100
@@ -36,7 +36,7 @@ SimpleKCM {
     property QtObject user
     property bool overrideImage: false
     property url oldImage
-
+    
     Connections {
         target: user
         function onApplyError(errorText) {
@@ -182,6 +182,17 @@ SimpleKCM {
                     changePassword.openAndClear()
                 }
             }
+            
+            QQC2.Button {
+                flat: false
+                visible: kcm.fingerprintModel.deviceFound
+                text: i18n("Fingerprint")
+                icon.name: "fingerprint-gui"
+                onClicked: {
+                    fingerprintDialog.account = user;
+                    fingerprintDialog.openAndClear();
+                }
+            }
 
             Item {
                 Layout.preferredHeight: deleteUser.height
@@ -452,4 +463,5 @@ SimpleKCM {
     }
 
     ChangePassword { id: changePassword; account: user }
+    FingerprintDialog { id: fingerprintDialog; account: user }
 }
diff -Npur plasma-desktop-5.20.5.orig/kcms/users/package/contents/ui/UserDetailsPage.qml.orig plasma-desktop-5.20.5/kcms/users/package/contents/ui/UserDetailsPage.qml.orig
--- plasma-desktop-5.20.5.orig/kcms/users/package/contents/ui/UserDetailsPage.qml.orig	1970-01-01 01:00:00.000000000 +0100
+++ plasma-desktop-5.20.5/kcms/users/package/contents/ui/UserDetailsPage.qml.orig	2021-01-05 13:13:03.000000000 +0100
@@ -0,0 +1,455 @@
+/*
+    Copyright 2019  Nicolas Fella <nicolas.fella@gmx.de>
+    Copyright 2020  Carson Black <uhhadd@gmail.com>
+
+    This library is free software; you can redistribute it and/or
+    modify it under the terms of the GNU Lesser General Public
+    License as published by the Free Software Foundation; either
+    version 2.1 of the License, or (at your option) version 3, or any
+    later version accepted by the membership of KDE e.V. (or its
+    successor approved by the membership of KDE e.V.), which shall
+    act as a proxy defined in Section 6 of version 3 of the license.
+
+    This library is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+    Lesser General Public License for more details.
+
+    You should have received a copy of the GNU Lesser General Public
+    License along with this library.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import QtQuick 2.6
+import QtQuick.Layouts 1.3
+import QtGraphicalEffects 1.12
+import QtQuick.Controls 2.5 as QQC2
+import QtQuick.Dialogs 1.3
+
+import org.kde.kcm 1.2
+import org.kde.kirigami 2.13 as Kirigami
+
+SimpleKCM {
+    id: usersDetailPage
+
+    title: user.realName
+
+    property QtObject user
+    property bool overrideImage: false
+    property url oldImage
+
+    Connections {
+        target: user
+        function onApplyError(errorText) {
+            errorMessage.visible = true
+            errorMessage.text = errorText
+        }
+    }
+
+
+    Connections {
+        target: kcm
+
+        onApply: {
+            errorMessage.visible = false
+            usersDetailPage.user.realName = realNametextField.text
+            usersDetailPage.user.email = emailTextField.text
+            usersDetailPage.user.name = userNameField.text
+            usersDetailPage.user.administrator = (usertypeBox.model[usertypeBox.currentIndex]["type"] == "administrator")
+            user.apply()
+            usersDetailPage.overrideImage = false
+            usersDetailPage.oldImage = ""
+        }
+
+        onReset: {
+            errorMessage.visible = false
+            realNametextField.text = usersDetailPage.user.realName
+            emailTextField.text = usersDetailPage.user.email
+            userNameField.text = usersDetailPage.user.name
+            usertypeBox.currentIndex = usersDetailPage.user.administrator ? 1 : 0
+            if (usersDetailPage.oldImage != "") {
+                usersDetailPage.overrideImage = false
+                usersDetailPage.user.face = usersDetailPage.oldImage
+            }
+        }
+    }
+
+    function resolvePending() {
+        let pending = false
+        let user = usersDetailPage.user
+        pending = pending || user.realName != realNametextField.text
+        pending = pending || user.email != emailTextField.text
+        pending = pending || user.name != userNameField.text
+        pending = pending || user.administrator != (usertypeBox.model[usertypeBox.currentIndex]["type"] == "administrator")
+        pending = pending || usersDetailPage.overrideImage
+        return pending
+    }
+
+    Component.onCompleted: {
+        kcm.needsSave = Qt.binding(resolvePending)
+    }
+
+    FileDialog {
+        id: fileDialog
+        title: i18n("Choose a picture")
+        folder: shortcuts.pictures
+        onAccepted: {
+            picturesSheet.close()
+            usersDetailPage.oldImage = usersDetailPage.user.face
+            usersDetailPage.user.face = fileDialog.fileUrl
+            usersDetailPage.overrideImage = true
+        }
+    }
+
+    ColumnLayout {
+        Kirigami.InlineMessage {
+            id: errorMessage
+            visible: false
+            type: Kirigami.MessageType.Error
+            Layout.fillWidth: true
+        }
+        RowLayout {
+            Layout.alignment: Qt.AlignHCenter
+            QQC2.RoundButton {
+                id: userPfp
+
+                readonly property int size: 6 * Kirigami.Units.gridUnit
+
+                implicitWidth: size
+                implicitHeight: size
+                flat: true
+
+                Kirigami.Avatar {
+                    source: usersDetailPage.user.face
+                    name: user.realName
+                    anchors {
+                        fill: parent
+                        margins: Kirigami.Units.smallSpacing
+                    }
+                }
+
+                Layout.alignment: Qt.AlignHCenter
+                Layout.maximumHeight: size + Kirigami.smallSpacing
+                Layout.minimumHeight: size + Kirigami.smallSpacing
+                Layout.maximumWidth: size + Kirigami.smallSpacing
+                Layout.minimumWidth: size + Kirigami.smallSpacing
+
+                onClicked: {
+                    picturesSheet.open()
+                    stackSwitcher.forceActiveFocus()
+                }
+            }
+            QQC2.TextField  {
+                id: realNametextField
+                focus: true
+                text: user.realName
+                placeholderText: i18nc("Example name", "John Doe")
+            }
+        }
+        Kirigami.FormLayout {
+            QQC2.TextField {
+                id: userNameField
+                focus: true
+                text: user.name
+                Kirigami.FormData.label: i18n("Username:")
+            }
+
+            QQC2.ComboBox {
+                id: usertypeBox
+
+                textRole: "label"
+                model: [
+                    { "type": "standard", "label": i18n("Standard") },
+                    { "type": "administrator", "label": i18n("Administrator") },
+                ]
+
+                Kirigami.FormData.label: i18n("Account type:")
+
+                currentIndex: user.administrator ? 1 : 0
+            }
+
+            QQC2.TextField {
+                id: emailTextField
+                focus: true
+                text: user.email
+                placeholderText: i18nc("Example email address", "john.doe@kde.org")
+                Kirigami.FormData.label: i18n("Email address:")
+            }
+
+            QQC2.Button {
+                text: i18n("Change Password")
+                onClicked: {
+                    changePassword.account = user
+                    changePassword.openAndClear()
+                }
+            }
+
+            Item {
+                Layout.preferredHeight: deleteUser.height
+            }
+
+            QQC2.Button {
+                id: deleteUser
+
+                enabled: !usersDetailPage.user.loggedIn && (!kcm.userModel.rowCount() < 2)
+
+                QQC2.Menu {
+                    id: deleteMenu
+                    modal: true
+                    QQC2.MenuItem {
+                        text: i18n("Delete files")
+                        icon.name: "edit-delete-shred"
+                        onClicked: {
+                            if (kcm.deleteUser(usersDetailPage.user.uid, true)) {
+                                kcm.pop()
+                            }
+                        }
+                    }
+                    QQC2.MenuItem {
+                        text: i18n("Keep files")
+                        icon.name: "document-multiple"
+                        onClicked: {
+                            if (kcm.deleteUser(usersDetailPage.user.uid, false)) {
+                                kcm.pop()
+                            }
+                        }
+                    }
+                }
+                text: i18n("Delete User...")
+                icon.name: "edit-delete"
+                onClicked: deleteMenu.open()
+            }
+        }
+    }
+
+    Kirigami.OverlaySheet {
+        id: picturesSheet
+        header: RowLayout {
+            Kirigami.Heading {
+                text: i18n("Change Avatar")
+            }
+        }
+        StackLayout {
+            id: stackSwitcher
+            Layout.preferredWidth: usersDetailPage.width - (Kirigami.Units.largeSpacing*4)
+            Keys.onEscapePressed: {
+                picturesSheet.close()
+                event.accepted = true
+            }
+            ColumnLayout {
+                id: cols
+                GridLayout {
+                    id: picturesColumn
+
+                    rowSpacing: Kirigami.Units.smallSpacing
+                    columnSpacing: Kirigami.Units.smallSpacing
+                    columns: {
+                        // subtract gridunit from stackswticher width to roughly compensate for slight overlap on tightly fit grids
+                        return Math.floor((stackSwitcher.width - Kirigami.Units.gridUnit) / ((Kirigami.Units.gridUnit * 6) + picturesColumn.columnSpacing))
+                    }
+
+                    Layout.fillWidth: true
+                    Layout.alignment: Qt.AlignHCenter
+
+                    QQC2.Button {
+                        Layout.preferredHeight: Kirigami.Units.gridUnit * 6
+                        Layout.preferredWidth: Layout.preferredHeight
+
+                        ColumnLayout {
+                            anchors.centerIn: parent
+
+                            Kirigami.Icon {
+                                width: Kirigami.Units.gridUnit * 4
+                                height: Kirigami.Units.gridUnit * 4
+                                source: "document-open"
+
+                                Layout.alignment: Qt.AlignHCenter
+                            }
+                            QQC2.Label {
+                                text: i18n("Choose File...")
+
+                                horizontalAlignment: Text.AlignHCenter
+                                verticalAlignment: Text.AlignBottom
+                                fontSizeMode: Text.HorizontalFit
+                                wrapMode: Text.Wrap
+                                Layout.fillWidth: true
+                                Layout.maximumWidth: Kirigami.Units.gridUnit * 5
+                            }
+                        }
+
+                        onClicked: fileDialog.open()
+                    }
+
+                    QQC2.Button {
+                        Layout.preferredHeight: Kirigami.Units.gridUnit * 6
+                        Layout.preferredWidth: Layout.preferredHeight
+
+                        Kirigami.Heading {
+                            anchors.centerIn: parent
+                            font.pixelSize: Kirigami.Units.gridUnit * 4
+                            text: kcm.initializeString(user.realName)
+                        }
+
+                        onClicked: stackSwitcher.currentIndex = 1
+                        // https://bugs.kde.org/show_bug.cgi?id=420439
+                        visible: false
+                    }
+
+                    Repeater {
+                        model: [
+                            "Artist Konqi.png",
+                            "Bookworm Konqi.png",
+                            "Boss Konqi.png",
+                            "Bug Catcher Konqi.png",
+                            "Card Shark Konqi.png",
+                            "Hacker Konqi.png",
+                            "Journalist Konqi.png",
+                            "Katie.png",
+                            "Konqi.png",
+                            "Mechanic Konqi.png",
+                            "Messenger Konqi.png",
+                            "Musician Konqi.png",
+                            "Office Worker Konqi.png",
+                            "PC Builder Konqi.png",
+                            "Scientist Konqi.png",
+                            "Teacher Konqi.png",
+                            "Virtual Reality Konqi.png",
+                            "photos/Air Balloon.png",
+                            "photos/Astronaut.png",
+                            "photos/Books.png",
+                            "photos/Brushes.png",
+                            "photos/Bulb.png",
+                            "photos/Car.png",
+                            "photos/Cat.png",
+                            "photos/Chameleon.png",
+                            "photos/Cocktail.png",
+                            "photos/Dog.png",
+                            "photos/Fish.png",
+                            "photos/Gamepad.png",
+                            "photos/Owl.png",
+                            "photos/Pancakes.png",
+                            "photos/Parrot.png",
+                            "photos/Pencils.png",
+                            "photos/Shuttle.png",
+                            "photos/Soccer.png",
+                            "photos/Sunflower.png",
+                            "photos/Sushi.png"
+                        ]
+                        QQC2.Button {
+                            Layout.preferredHeight: Kirigami.Units.gridUnit * 6
+                            Layout.preferredWidth: Layout.preferredHeight
+
+                            QQC2.ToolTip.delay: Kirigami.Units.toolTipDelay
+                            QQC2.ToolTip.text: modelData
+                            Accessible.name: modelData
+
+                            Image {
+                                id: imgDelegate
+                                smooth: true
+                                mipmap: true
+
+                                anchors.fill: parent
+                                source: "../img/%1".arg(modelData)
+
+                                Accessible.ignored: true
+                            }
+
+                            onClicked: {
+                                picturesSheet.close()
+                                usersDetailPage.oldImage = usersDetailPage.user.face
+                                usersDetailPage.user.face = imgDelegate.source
+                                usersDetailPage.overrideImage = true
+                            }
+                        }
+                    }
+                }
+            }
+            ColumnLayout {
+                id: initialPictures
+                GridLayout {
+                    id: initialsColumn
+
+                    rowSpacing: Kirigami.Units.smallSpacing
+                    columnSpacing: Kirigami.Units.smallSpacing
+                    columns: {
+                        return Math.floor((stackSwitcher.width) / (Kirigami.Units.gridUnit * 6))
+                    }
+
+                    Layout.fillWidth: true
+                    Layout.alignment: Qt.AlignHCenter
+
+                    QQC2.Button {
+                        Layout.preferredHeight: Kirigami.Units.gridUnit * 6
+                        Layout.preferredWidth: Layout.preferredHeight
+
+                        ColumnLayout {
+                            anchors.centerIn: parent
+
+                            Kirigami.Icon {
+                                width: Kirigami.Units.gridUnit * 4
+                                height: Kirigami.Units.gridUnit * 4
+                                source: "go-previous"
+
+                                Layout.alignment: Qt.AlignHCenter
+                            }
+                        }
+
+                        onClicked: stackSwitcher.currentIndex = 0
+                    }
+
+                    Repeater {
+                        model: [
+                            {"name": i18n("Feisty Flamingo"),  "color": "#E93A9A", "dark": true},
+                            {"name": i18n("Dragon's Fruit"),   "color": "#E93D58", "dark": true},
+                            {"name": i18n("Sweet Potato"),     "color": "#E9643A", "dark": true},
+                            {"name": i18n("Ambient Amber"),    "color": "#EF973C", "dark": false},
+                            {"name": i18n("Sparkle Sunbeam"),  "color": "#E8CB2D", "dark": false},
+                            {"name": i18n("Lemon-Lime"),       "color": "#B6E521", "dark": false},
+                            {"name": i18n("Verdant Charm"),    "color": "#3DD425", "dark": false},
+                            {"name": i18n("Mellow Meadow"),    "color": "#00D485", "dark": false},
+                            {"name": i18n("Tepid Teal"),       "color": "#00D3B8", "dark": false},
+                            {"name": i18n("Plasma Blue"),      "color": "#3DAEE9", "dark": true},
+                            {"name": i18n("Pon Purple"),       "color": "#B875DC", "dark": true},
+                            {"name": i18n("Bajo Purple"),      "color": "#926EE4", "dark": true},
+                            {"name": i18n("Burnt Charcoal"),   "color": "#232629", "dark": true},
+                            {"name": i18n("Paper Perfection"), "color": "#EEF1F5", "dark": false},
+                            {"name": i18n("Cafétera Brown"),   "color": "#CB775A", "dark": false},
+                            {"name": i18n("Rich Hardwood"),    "color": "#6A250E", "dark": true}
+                        ]
+                        delegate: QQC2.Button {
+                            Layout.preferredHeight: Kirigami.Units.gridUnit * 6
+                            Layout.preferredWidth: Layout.preferredHeight
+
+                            Rectangle {
+                                id: colourRectangle
+
+                                anchors.fill: parent
+                                anchors.margins: Kirigami.Units.smallSpacing
+                                color: modelData.color
+                                Kirigami.Heading {
+                                    anchors.centerIn: parent
+                                    color: modelData.dark ? "white" : "black"
+                                    font.pixelSize: Kirigami.Units.gridUnit * 4
+                                    text: kcm.initializeString(user.realName)
+                                }
+                            }
+
+                            onClicked: {
+                                colourRectangle.grabToImage(function(result) {
+                                    picturesSheet.close()
+                                    let uri = kcm.plonkImageInTempfile(result.image)
+                                    if (uri != "") {
+                                        usersDetailPage.oldImage = usersDetailPage.user.face
+                                        usersDetailPage.user.face = uri
+                                        usersDetailPage.overrideImage = true
+                                    }
+                                })
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    ChangePassword { id: changePassword; account: user }
+}
diff -Npur plasma-desktop-5.20.5.orig/kcms/users/src/CMakeLists.txt plasma-desktop-5.20.5/kcms/users/src/CMakeLists.txt
--- plasma-desktop-5.20.5.orig/kcms/users/src/CMakeLists.txt	2021-01-05 13:13:03.000000000 +0100
+++ plasma-desktop-5.20.5/kcms/users/src/CMakeLists.txt	2021-01-23 14:14:52.784039095 +0100
@@ -2,6 +2,8 @@ set(kcm_users_SRCS
     kcm.cpp
     user.cpp
     usermodel.cpp
+    fprintdevice.cpp
+    fingerprintmodel.cpp
 )
 
 set_source_files_properties(org.freedesktop.Accounts.xml
@@ -10,6 +12,12 @@ set_source_files_properties(org.freedesk
 set_source_files_properties(org.freedesktop.Accounts.User.xml
                         PROPERTIES NO_NAMESPACE TRUE)
 
+set_source_files_properties(net.reactivated.Fprint.Device.xml
+                        PROPERTIES NO_NAMESPACE TRUE)
+
+set_source_files_properties(net.reactivated.Fprint.Manager.xml
+                        PROPERTIES NO_NAMESPACE TRUE)
+                        
 qt5_add_dbus_interface(kcm_users_SRCS
     org.freedesktop.Accounts.xml
     accounts_interface
@@ -20,6 +28,16 @@ qt5_add_dbus_interface(kcm_users_SRCS
     user_interface
 )
 
+qt5_add_dbus_interface(kcm_users_SRCS
+    net.reactivated.Fprint.Device.xml
+    fprint_device_interface
+)
+
+qt5_add_dbus_interface(kcm_users_SRCS
+    net.reactivated.Fprint.Manager.xml
+    fprint_manager_interface
+)
+
 set(login1_manager_xml org.freedesktop.login1.Manager.xml)
 set_source_files_properties(${login1_manager_xml} PROPERTIES INCLUDE "usersessions.h")
 qt5_add_dbus_interface(kcm_users_SRCS
diff -Npur plasma-desktop-5.20.5.orig/kcms/users/src/fingerprintmodel.cpp plasma-desktop-5.20.5/kcms/users/src/fingerprintmodel.cpp
--- plasma-desktop-5.20.5.orig/kcms/users/src/fingerprintmodel.cpp	1970-01-01 01:00:00.000000000 +0100
+++ plasma-desktop-5.20.5/kcms/users/src/fingerprintmodel.cpp	2021-01-23 14:04:18.477355122 +0100
@@ -0,0 +1,323 @@
+/*
+    Copyright 2020  Devin Lin <espidev@gmail.com>
+
+    This library is free software; you can redistribute it and/or
+    modify it under the terms of the GNU Lesser General Public
+    License as published by the Free Software Foundation; either
+    version 2.1 of the License, or (at your option) version 3, or any
+    later version accepted by the membership of KDE e.V. (or its
+    successor approved by the membership of KDE e.V.), which shall
+    act as a proxy defined in Section 6 of version 3 of the license.
+
+    This library is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+    Lesser General Public License for more details.
+
+    You should have received a copy of the GNU Lesser General Public
+    License along with this library.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <KLocalizedString>
+
+#include "fingerprintmodel.h"
+
+#include "fprint_device_interface.h"
+#include "fprint_manager_interface.h"
+
+FingerprintModel::FingerprintModel(QObject* parent)
+    : QObject(parent)
+    , m_managerDbusInterface(new NetReactivatedFprintManagerInterface(QStringLiteral("net.reactivated.Fprint"), QStringLiteral("/net/reactivated/Fprint/Manager"), QDBusConnection::systemBus(), this))
+{
+    auto reply = m_managerDbusInterface->GetDefaultDevice();
+    reply.waitForFinished();
+    
+    if (reply.isError()) {
+        qDebug() << reply.error().message();
+        setCurrentError(reply.error().message());
+        return;
+    }
+
+    QDBusObjectPath path = reply.value();
+    m_device = new FprintDevice(path, this);
+    
+    connect(m_device, &FprintDevice::enrollCompleted, this, &FingerprintModel::handleEnrollCompleted);
+    connect(m_device, &FprintDevice::enrollStagePassed, this, &FingerprintModel::handleEnrollStagePassed);
+    connect(m_device, &FprintDevice::enrollRetryStage, this, &FingerprintModel::handleEnrollRetryStage);
+    connect(m_device, &FprintDevice::enrollFailed, this, &FingerprintModel::handleEnrollFailed);
+}
+
+FingerprintModel::~FingerprintModel()
+{
+    if (m_device) { // just in case device is claimed
+        m_device->stopEnrolling();
+        m_device->release();
+    }
+}
+
+QString FingerprintModel::scanType()
+{
+    return !m_device ? "" : m_device->scanType();
+}
+
+QString FingerprintModel::currentError()
+{
+    return m_currentError;
+}
+
+void FingerprintModel::setCurrentError(QString error)
+{
+    if (m_currentError != error) {
+        m_currentError = error;
+        Q_EMIT currentErrorChanged();
+    }
+}
+
+QString FingerprintModel::enrollFeedback()
+{
+    return m_enrollFeedback;
+}
+
+void FingerprintModel::setEnrollFeedback(QString feedback)
+{
+    m_enrollFeedback = feedback;
+    Q_EMIT enrollFeedbackChanged();
+}
+
+bool FingerprintModel::currentlyEnrolling()
+{
+    return m_currentlyEnrolling;
+}
+
+bool FingerprintModel::deviceFound()
+{
+    return m_device != nullptr;
+}
+
+double FingerprintModel::enrollProgress()
+{
+    if (!deviceFound()) {
+        return 0;
+    }
+    return (m_device->numOfEnrollStages() == 0) ? 1 : ((double) m_enrollStage) / m_device->numOfEnrollStages();
+}
+
+void FingerprintModel::setEnrollStage(int stage)
+{
+    m_enrollStage = stage;
+    Q_EMIT enrollProgressChanged();
+}
+
+FingerprintModel::DialogState FingerprintModel::dialogState()
+{
+    return m_dialogState;
+}
+
+void FingerprintModel::setDialogState(DialogState dialogState)
+{
+    m_dialogState = dialogState;
+    Q_EMIT dialogStateChanged();
+}
+
+void FingerprintModel::switchUser(QString username)
+{
+    m_username = username;
+    
+    if (deviceFound()) {
+        stopEnrolling(); // stop enrolling if ongoing
+        m_device->release(); // release from old user
+        
+        Q_EMIT enrolledFingerprintsChanged();
+    }
+}
+
+bool FingerprintModel::claimDevice()
+{
+    if (!deviceFound()) {
+        return false;
+    }
+    
+    QDBusError error = m_device->claim(m_username);
+    if (error.isValid() && error.name() != "net.reactivated.Fprint.Error.AlreadyInUse") {
+        qDebug() << "error claiming:" << error.message();
+        setCurrentError(error.message());
+        return false;
+    }
+    return true;
+}
+
+void FingerprintModel::startEnrolling(QString finger)
+{
+    if (!deviceFound()) {
+        setCurrentError(i18n("No fingerprint device found."));
+        setDialogState(DialogState::FingerprintList);
+        return;
+    }
+
+    setEnrollStage(0);
+    setEnrollFeedback({});
+    
+    // claim device for user
+    if (!claimDevice()) {
+        setDialogState(DialogState::FingerprintList);
+        return;
+    }
+    
+    QDBusError error = m_device->startEnrolling(finger);
+    if (error.isValid()) {
+        qDebug() << "error start enrolling:" << error.message();
+        setCurrentError(error.message());
+        m_device->release();
+        setDialogState(DialogState::FingerprintList);
+        return;
+    }
+    
+    m_currentlyEnrolling = true;
+    Q_EMIT currentlyEnrollingChanged();
+    
+    setDialogState(DialogState::Enrolling);
+}
+
+void FingerprintModel::stopEnrolling()
+{
+    setDialogState(DialogState::FingerprintList);
+    if (m_currentlyEnrolling) {
+        m_currentlyEnrolling = false;
+        Q_EMIT currentlyEnrollingChanged();
+        
+        QDBusError error = m_device->stopEnrolling();
+        if (error.isValid()) {
+            qDebug() << "error stop enrolling:" << error.message();
+            setCurrentError(error.message());
+            return;
+        }
+        m_device->release();
+    }
+}
+
+void FingerprintModel::clearFingerprints()
+{
+    // claim for user
+    if (!claimDevice()) {
+        return;
+    }
+ 
+    QDBusError error = m_device->deleteEnrolledFingers();
+    if (error.isValid()) {
+        qDebug() << "error clearing fingerprints:" << error.message();
+        setCurrentError(error.message());
+    }
+    
+    // release from user
+    error = m_device->release();
+    if (error.isValid()) {
+        qDebug() << "error releasing:" << error.message();
+        setCurrentError(error.message());
+    }
+    
+    Q_EMIT enrolledFingerprintsChanged();
+}
+
+QStringList FingerprintModel::enrolledFingerprintsRaw()
+{
+    if (deviceFound()) {
+        QDBusPendingReply reply = m_device->listEnrolledFingers(m_username);
+        reply.waitForFinished();
+        if (reply.isError()) {
+            // ignore net.reactivated.Fprint.Error.NoEnrolledPrints, as it shows up when there are no fingerprints
+            if (reply.error().name() != "net.reactivated.Fprint.Error.NoEnrolledPrints") {
+                qDebug() << "error listing enrolled fingers:" << reply.error().message();
+                setCurrentError(reply.error().message());
+            }
+            return QStringList();
+        }
+        return reply.value();
+    } else {
+        setCurrentError(i18n("No fingerprint device found."));
+        setDialogState(DialogState::FingerprintList);
+        return QStringList();
+    }
+}
+
+QVariantList FingerprintModel::enrolledFingerprints()
+{
+    // convert fingers list to qlist of Finger objects
+    QVariantList fingers;
+    for (QString &finger : enrolledFingerprintsRaw()) {
+        for (Finger *storedFinger : FINGERS) {
+            if (storedFinger->internalName() == finger) {
+                fingers.append(QVariant::fromValue(storedFinger));
+                break;
+            }
+        }
+    }
+    return fingers;
+}
+
+QVariantList FingerprintModel::availableFingersToEnroll()
+{
+    QVariantList list; 
+    QStringList enrolled = enrolledFingerprintsRaw();
+    
+    // add fingerprints to list that are not in the enrolled list
+    for (Finger *finger : FINGERS) {
+        if (!enrolledFingerprintsRaw().contains(finger->internalName())) {
+            list.append(QVariant::fromValue(finger));
+        }
+    }
+    return list;
+}
+
+void FingerprintModel::handleEnrollCompleted()
+{
+    setEnrollStage(m_device->numOfEnrollStages());
+    setEnrollFeedback({});
+    Q_EMIT enrolledFingerprintsChanged();
+    Q_EMIT scanComplete();
+    
+    // stopEnrolling not called, as it is triggered only when the "complete" button is pressed
+    // (only change dialog state change after button is pressed)
+    setDialogState(DialogState::EnrollComplete);
+}
+
+void FingerprintModel::handleEnrollStagePassed()
+{
+    setEnrollStage(m_enrollStage + 1);
+    setEnrollFeedback({});
+    Q_EMIT scanSuccess();
+    qDebug() << "fingerprint enroll stage pass:" << enrollProgress();
+}
+
+void FingerprintModel::handleEnrollRetryStage(QString feedback)
+{
+    Q_EMIT scanFailure();
+    if (feedback == "enroll-retry-scan") {
+        setEnrollFeedback(i18n("Retry scanning your finger."));
+    } else if (feedback == "enroll-swipe-too-short") {
+        setEnrollFeedback(i18n("Swipe too short. Try again."));
+    } else if (feedback == "enroll-finger-not-centered") {
+        setEnrollFeedback(i18n("Finger not centered on the reader. Try again."));
+    } else if (feedback == "enroll-remove-and-retry") {
+        setEnrollFeedback(i18n("Remove your finger from the reader, and try again."));
+    }
+    qDebug() << "fingerprint enroll stage fail:" << feedback;
+}
+
+void FingerprintModel::handleEnrollFailed(QString error)
+{
+    if (error == "enroll-failed") {
+        setCurrentError(i18n("Fingerprint enrollment has failed."));
+        stopEnrolling();
+    } else if (error == "enroll-data-full") {
+        setCurrentError(i18n("There is no space left for this device, delete other fingerprints to continue."));
+        stopEnrolling();
+    } else if (error == "enroll-disconnected") {
+        setCurrentError(i18n("The device was disconnected."));
+        m_currentlyEnrolling = false;
+        Q_EMIT currentlyEnrollingChanged();
+        setDialogState(DialogState::FingerprintList);
+    } else if (error == "enroll-unknown-error") {
+        setCurrentError(i18n("An unknown error has occurred."));
+        stopEnrolling();
+    }
+}
diff -Npur plasma-desktop-5.20.5.orig/kcms/users/src/fingerprintmodel.h plasma-desktop-5.20.5/kcms/users/src/fingerprintmodel.h
--- plasma-desktop-5.20.5.orig/kcms/users/src/fingerprintmodel.h	1970-01-01 01:00:00.000000000 +0100
+++ plasma-desktop-5.20.5/kcms/users/src/fingerprintmodel.h	2021-01-23 14:04:18.477355122 +0100
@@ -0,0 +1,153 @@
+/*
+    Copyright 2020  Devin Lin <espidev@gmail.com>
+
+    This library is free software; you can redistribute it and/or
+    modify it under the terms of the GNU Lesser General Public
+    License as published by the Free Software Foundation; either
+    version 2.1 of the License, or (at your option) version 3, or any
+    later version accepted by the membership of KDE e.V. (or its
+    successor approved by the membership of KDE e.V.), which shall
+    act as a proxy defined in Section 6 of version 3 of the license.
+
+    This library is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+    Lesser General Public License for more details.
+
+    You should have received a copy of the GNU Lesser General Public
+    License along with this library.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "fprintdevice.h"
+#include <qobjectdefs.h>
+
+#include <KLocalizedString>
+
+#include "fprint_device_interface.h"
+#include "fprint_manager_interface.h"
+
+class Finger : public QObject {
+    Q_OBJECT
+    Q_PROPERTY(QString internalName READ internalName CONSTANT)
+    Q_PROPERTY(QString friendlyName READ friendlyName CONSTANT)
+    
+public:
+    explicit Finger(const Finger &finger, QObject* parent = nullptr)
+    : QObject(parent)
+    , m_internalName(finger.internalName())
+    , m_friendlyName(finger.friendlyName())
+    {}
+
+    explicit Finger(QString internalName = "", QString friendlyName = "", QObject* parent = nullptr)
+    : QObject(parent)
+    , m_internalName(internalName)
+    , m_friendlyName(friendlyName)
+    {}
+    
+    QString internalName() const
+    {
+        return m_internalName;
+    }
+    QString friendlyName() const
+    {
+        return m_friendlyName;
+    }
+private:
+    QString m_internalName = "", m_friendlyName = "";
+};
+
+class FingerprintModel : public QObject
+{
+    Q_OBJECT
+    Q_PROPERTY(QString scanType READ scanType CONSTANT)
+    Q_PROPERTY(QString currentError READ currentError WRITE setCurrentError NOTIFY currentErrorChanged) // error for ui to display
+    Q_PROPERTY(QString enrollFeedback READ enrollFeedback WRITE setEnrollFeedback NOTIFY enrollFeedbackChanged)
+    Q_PROPERTY(QVariantList enrolledFingerprints READ enrolledFingerprints NOTIFY enrolledFingerprintsChanged)
+    Q_PROPERTY(QVariantList availableFingersToEnroll READ availableFingersToEnroll NOTIFY enrolledFingerprintsChanged)
+    Q_PROPERTY(bool deviceFound READ deviceFound NOTIFY devicesFoundChanged)
+    Q_PROPERTY(bool currentlyEnrolling READ currentlyEnrolling NOTIFY currentlyEnrollingChanged)
+    Q_PROPERTY(double enrollProgress READ enrollProgress NOTIFY enrollProgressChanged)
+    Q_PROPERTY(DialogState dialogState READ dialogState WRITE setDialogState NOTIFY dialogStateChanged)
+    
+public:
+    explicit FingerprintModel(QObject* parent = nullptr);
+    ~FingerprintModel();
+
+    enum DialogState {
+        FingerprintList,
+        PickFinger,
+        Enrolling,
+        EnrollComplete,
+    };
+    Q_ENUM(DialogState)
+    
+    const QList<Finger *> FINGERS = {
+        new Finger("right-index-finger", i18n("Right index finger"), this),
+        new Finger("right-middle-finger", i18n("Right middle finger"), this),
+        new Finger("right-ring-finger", i18n("Right ring finger"), this),
+        new Finger("right-little-finger", i18n("Right little finger"), this),
+        new Finger("right-thumb", i18n("Right thumb"), this),
+        new Finger("left-index-finger", i18n("Left index finger"), this),
+        new Finger("left-middle-finger", i18n("Left middle finger"), this),
+        new Finger("left-ring-finger", i18n("Left ring finger"), this),
+        new Finger("left-little-finger", i18n("Left little finger"), this),
+        new Finger("left-thumb", i18n("Left thumb"), this)
+    };
+    
+    Q_INVOKABLE void switchUser(QString username);
+    bool claimDevice();
+    
+    Q_INVOKABLE void startEnrolling(QString finger);
+    Q_INVOKABLE void stopEnrolling();
+    Q_INVOKABLE void clearFingerprints();
+    
+    QStringList enrolledFingerprintsRaw();
+    QVariantList enrolledFingerprints();
+    QVariantList availableFingersToEnroll();
+    
+    QString scanType();
+    QString currentError();
+    void setCurrentError(QString error);
+    QString enrollFeedback();
+    void setEnrollFeedback(QString feedback);
+    bool currentlyEnrolling();
+    bool deviceFound();
+    double enrollProgress();
+    void setEnrollStage(int stage);
+    DialogState dialogState();
+    void setDialogState(DialogState dialogState);
+    
+public Q_SLOTS:
+    void handleEnrollCompleted();
+    void handleEnrollStagePassed();
+    void handleEnrollRetryStage(QString feedback);
+    void handleEnrollFailed(QString error);
+
+Q_SIGNALS:
+    void currentErrorChanged();
+    void enrollFeedbackChanged();
+    void enrolledFingerprintsChanged();
+    void devicesFoundChanged();
+    void currentlyEnrollingChanged();
+    void enrollProgressChanged();
+    void dialogStateChanged();
+
+    void scanComplete();
+    void scanSuccess();
+    void scanFailure();
+    
+private:
+    QString m_username; // set to "" if it is the currently logged in user
+    QString m_currentError, m_enrollFeedback;
+    
+    DialogState m_dialogState = DialogState::FingerprintList;
+    
+    bool m_currentlyEnrolling = false;
+    int m_enrollStage = 0;
+    
+    FprintDevice *m_device = nullptr;
+    
+    NetReactivatedFprintManagerInterface *m_managerDbusInterface;
+};
diff -Npur plasma-desktop-5.20.5.orig/kcms/users/src/fprintdevice.cpp plasma-desktop-5.20.5/kcms/users/src/fprintdevice.cpp
--- plasma-desktop-5.20.5.orig/kcms/users/src/fprintdevice.cpp	1970-01-01 01:00:00.000000000 +0100
+++ plasma-desktop-5.20.5/kcms/users/src/fprintdevice.cpp	2021-01-23 14:04:18.477355122 +0100
@@ -0,0 +1,96 @@
+/*
+    Copyright 2020  Devin Lin <espidev@gmail.com>
+
+    This library is free software; you can redistribute it and/or
+    modify it under the terms of the GNU Lesser General Public
+    License as published by the Free Software Foundation; either
+    version 2.1 of the License, or (at your option) version 3, or any
+    later version accepted by the membership of KDE e.V. (or its
+    successor approved by the membership of KDE e.V.), which shall
+    act as a proxy defined in Section 6 of version 3 of the license.
+
+    This library is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+    Lesser General Public License for more details.
+
+    You should have received a copy of the GNU Lesser General Public
+    License along with this library.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "fprintdevice.h"
+
+#include "fprint_device_interface.h"
+#include "fprint_manager_interface.h"
+
+FprintDevice::FprintDevice(QDBusObjectPath path, QObject* parent)
+    : QObject(parent)
+    , m_dbusInterface(new NetReactivatedFprintDeviceInterface(QStringLiteral("net.reactivated.Fprint"), path.path(), QDBusConnection::systemBus(), this))
+{
+    connect(m_dbusInterface, &NetReactivatedFprintDeviceInterface::EnrollStatus, this, &FprintDevice::enrollStatus);
+}
+
+QDBusPendingReply<QStringList> FprintDevice::listEnrolledFingers(const QString &username)
+{
+    return m_dbusInterface->ListEnrolledFingers(username);
+}
+
+QDBusError FprintDevice::claim(const QString &username)
+{
+    auto reply = m_dbusInterface->Claim(username);
+    reply.waitForFinished();
+    return reply.error();
+}
+
+QDBusError FprintDevice::release()
+{
+    auto reply = m_dbusInterface->Release();
+    reply.waitForFinished();
+    return reply.error();
+}
+
+QDBusError FprintDevice::deleteEnrolledFingers()
+{
+    auto reply = m_dbusInterface->DeleteEnrolledFingers2();
+    reply.waitForFinished();
+    return reply.error();
+}
+
+QDBusError FprintDevice::startEnrolling(const QString &finger)
+{
+    auto reply = m_dbusInterface->EnrollStart(finger);
+    reply.waitForFinished();
+    return reply.error();
+}
+
+QDBusError FprintDevice::stopEnrolling()
+{
+    auto reply = m_dbusInterface->EnrollStop();
+    reply.waitForFinished();
+    return reply.error();
+}
+
+void FprintDevice::enrollStatus(QString result, bool done)
+{
+    if (result == "enroll-completed") {
+        Q_EMIT enrollCompleted();
+    } else if (result == "enroll-failed" || result == "enroll-data-full" || result == "enroll-disconnected" || result == "enroll-unknown-error") {
+        Q_EMIT enrollFailed(result);
+    } else if (result == "enroll-stage-passed") {
+        Q_EMIT enrollStagePassed();
+    } else if (result == "enroll-retry-scan" || result == "enroll-swipe-too-short" || result == "enroll-finger-not-centered" || result == "enroll-remove-and-retry") {
+        Q_EMIT enrollRetryStage(result);
+    }
+}
+
+int FprintDevice::numOfEnrollStages()
+{
+    return m_dbusInterface->num_enroll_stages();
+}
+
+QString FprintDevice::scanType()
+{
+    return m_dbusInterface->scan_type();
+}
+
+
diff -Npur plasma-desktop-5.20.5.orig/kcms/users/src/fprintdevice.h plasma-desktop-5.20.5/kcms/users/src/fprintdevice.h
--- plasma-desktop-5.20.5.orig/kcms/users/src/fprintdevice.h	1970-01-01 01:00:00.000000000 +0100
+++ plasma-desktop-5.20.5/kcms/users/src/fprintdevice.h	2021-01-23 14:04:18.477355122 +0100
@@ -0,0 +1,58 @@
+/*
+    Copyright 2020  Devin Lin <espidev@gmail.com>
+
+    This library is free software; you can redistribute it and/or
+    modify it under the terms of the GNU Lesser General Public
+    License as published by the Free Software Foundation; either
+    version 2.1 of the License, or (at your option) version 3, or any
+    later version accepted by the membership of KDE e.V. (or its
+    successor approved by the membership of KDE e.V.), which shall
+    act as a proxy defined in Section 6 of version 3 of the license.
+
+    This library is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+    Lesser General Public License for more details.
+
+    You should have received a copy of the GNU Lesser General Public
+    License along with this library.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "fprint_device_interface.h"
+#include "fprint_manager_interface.h"
+
+class FprintDevice : public QObject
+{
+    Q_OBJECT
+
+public:
+    explicit FprintDevice(QDBusObjectPath path, QObject* parent = nullptr);
+
+    QDBusPendingReply<QStringList> listEnrolledFingers(const QString &username);
+    
+    QDBusError claim(const QString &username);
+    QDBusError release();
+    
+    QDBusError deleteEnrolledFingers();
+    QDBusError startEnrolling(const QString &finger);
+    QDBusError stopEnrolling();
+
+    int numOfEnrollStages();
+    QString scanType();
+    
+public Q_SLOTS:
+    void enrollStatus(QString result, bool done);
+    
+Q_SIGNALS:
+    void enrollCompleted();
+    void enrollStagePassed();
+    void enrollRetryStage(QString feedback);
+    void enrollFailed(QString error);
+    
+private:
+    QString m_username;
+    NetReactivatedFprintDeviceInterface* m_dbusInterface;
+};
+
diff -Npur plasma-desktop-5.20.5.orig/kcms/users/src/kcm.cpp plasma-desktop-5.20.5/kcms/users/src/kcm.cpp
--- plasma-desktop-5.20.5.orig/kcms/users/src/kcm.cpp	2021-01-05 13:13:03.000000000 +0100
+++ plasma-desktop-5.20.5/kcms/users/src/kcm.cpp	2021-01-23 14:04:18.477355122 +0100
@@ -36,6 +36,7 @@
 #include <QTemporaryFile>
 
 #include "user.h"
+#include "fingerprintmodel.h"
 #include "accounts_interface.h"
 
 Q_LOGGING_CATEGORY(kcm_users, "kcm_users")
@@ -46,17 +47,21 @@ K_PLUGIN_FACTORY_WITH_JSON(KCMUserFactor
 KCMUser::KCMUser(QObject *parent, const QVariantList &args)
     : KQuickAddons::ConfigModule(parent, args),
       m_dbusInterface(new OrgFreedesktopAccountsInterface(QStringLiteral("org.freedesktop.Accounts"), QStringLiteral("/org/freedesktop/Accounts"), QDBusConnection::systemBus(), this)),
-      m_model(new UserModel(this))
+      m_model(new UserModel(this)),
+      m_fingerprintModel(new FingerprintModel(this))
 {
     KAboutData* about = new KAboutData(QStringLiteral("kcm_users"), i18n("Manage user accounts"),
                                        QStringLiteral("0.1"), QString(), KAboutLicense::GPL);
     about->addAuthor(i18n("Nicolas Fella"), QString(), QStringLiteral("nicolas.fella@gmx.de"));
     about->addAuthor(i18n("Carson Black"), QString(), QStringLiteral("uhhadd@gmail.com"));
+    about->addAuthor(i18n("Devin Lin"), QString(), QStringLiteral("espidev@gmail.com"));
     setAboutData(about);
     setButtons(Apply);
     auto font = QApplication::font("QLabel");
     auto fm = QFontMetrics(font);
     setColumnWidth(fm.capHeight()*30);
+    
+    qmlRegisterType<Finger>("FingerprintModel", 1, 0, "Finger");
 }
 
 bool KCMUser::createUser(const QString& name, const QString& realName, const QString& password, bool isAdmin)
diff -Npur plasma-desktop-5.20.5.orig/kcms/users/src/kcm.h plasma-desktop-5.20.5/kcms/users/src/kcm.h
--- plasma-desktop-5.20.5.orig/kcms/users/src/kcm.h	2021-01-05 13:13:03.000000000 +0100
+++ plasma-desktop-5.20.5/kcms/users/src/kcm.h	2021-01-23 14:04:18.467355121 +0100
@@ -22,6 +22,7 @@
 
 #include <KQuickAddons/ConfigModule>
 #include "usermodel.h"
+#include "fingerprintmodel.h"
 
 class OrgFreedesktopAccountsInterface;
 
@@ -31,10 +32,12 @@ class KCMUser : public KQuickAddons::Con
 {
     Q_OBJECT
     Q_PROPERTY(UserModel* userModel MEMBER m_model CONSTANT)
-
+    Q_PROPERTY(FingerprintModel* fingerprintModel MEMBER m_fingerprintModel CONSTANT)
+    
 private:
     OrgFreedesktopAccountsInterface* m_dbusInterface;
     UserModel* m_model;
+    FingerprintModel* m_fingerprintModel;
 
 public:
     KCMUser(QObject *parent = nullptr, const QVariantList &args = QVariantList());
diff -Npur plasma-desktop-5.20.5.orig/kcms/users/src/net.reactivated.Fprint.Device.xml plasma-desktop-5.20.5/kcms/users/src/net.reactivated.Fprint.Device.xml
--- plasma-desktop-5.20.5.orig/kcms/users/src/net.reactivated.Fprint.Device.xml	1970-01-01 01:00:00.000000000 +0100
+++ plasma-desktop-5.20.5/kcms/users/src/net.reactivated.Fprint.Device.xml	2021-01-23 14:04:18.474021787 +0100
@@ -0,0 +1,599 @@
+<!DOCTYPE node PUBLIC
+"-//freedesktop//DTD D-BUS Object Introspection 1.0//EN"
+"http://www.freedesktop.org/standards/dbus/1.0/introspect.dtd" [
+<!ENTITY ERROR_CLAIM_DEVICE "net.reactivated.Fprint.Error.ClaimDevice">
+<!ENTITY ERROR_ALREADY_IN_USE "net.reactivated.Fprint.Error.AlreadyInUse">
+<!ENTITY ERROR_INTERNAL "net.reactivated.Fprint.Error.Internal">
+<!ENTITY ERROR_PERMISSION_DENIED "net.reactivated.Fprint.Error.PermissionDenied">
+<!ENTITY ERROR_NO_ENROLLED_PRINTS "net.reactivated.Fprint.Error.NoEnrolledPrints">
+<!ENTITY ERROR_NO_ACTION_IN_PROGRESS "net.reactivated.Fprint.Error.NoActionInProgress">
+<!ENTITY ERROR_INVALID_FINGERNAME "net.reactivated.Fprint.Error.InvalidFingername">
+]>
+
+<node name="/" xmlns:doc="http://www.freedesktop.org/dbus/1.0/doc.dtd">
+	<interface name="net.reactivated.Fprint.Device">
+		<annotation name="org.freedesktop.DBus.GLib.CSymbol"
+			value="fprint_device" />
+
+		<doc:doc>
+			<doc:title id="polkit-integration">
+				PolicyKit integration
+			</doc:title>
+			<doc:para>
+				fprintd uses PolicyKit to check whether users are allowed to access fingerprint data, or the
+				fingerprint readers itself.
+				<doc:list>
+					<doc:item>
+						<doc:term>net.reactivated.fprint.device.verify</doc:term>
+						<doc:definition>
+							Whether the user is allowed to verify fingers against saved fingerprints.
+						</doc:definition>
+					</doc:item>
+					<doc:item>
+						<doc:term>net.reactivated.fprint.device.enroll</doc:term>
+						<doc:definition>
+							Whether the user is allowed to enroll new fingerprints.
+						</doc:definition>
+					</doc:item>
+					<doc:item>
+						<doc:term>net.reactivated.fprint.device.setusername</doc:term>
+						<doc:definition>
+							Whether the user is allowed to query, verify, or enroll fingerprints for users other than itself.
+						</doc:definition>
+					</doc:item>
+				</doc:list>
+			</doc:para>
+
+			<doc:title id="usernames">
+				Usernames
+			</doc:title>
+			<doc:para>
+				When a username argument is used for a method, a PolicyKit check is done on the
+				<doc:tt>net.reactivated.fprint.device.setusername</doc:tt> PolicyKit
+				action to see whether the user the client is running as is allowed to access data from other users.
+			</doc:para>
+			<doc:para>
+				By default, only root is allowed to access fingerprint data for users other than itself. For a normal user,
+				it is recommended that you use an empty string for the username, which will mean "the client the user is
+				running as".
+			</doc:para>
+			<doc:para>
+				See <doc:ref type="description" to="polkit-integration">PolicyKit integration</doc:ref>.
+			</doc:para>
+
+			<doc:title id="fingerprint-names">
+				Fingerprint names
+			</doc:title>
+			<doc:para>
+				When a finger name argument is used for a method, it refers to either a single finger, or
+				"any" finger. See the list of possible values below:
+				<doc:list>
+					<doc:item>
+						<doc:term>left-thumb</doc:term>
+						<doc:definition>
+							Left thumb
+						</doc:definition>
+					</doc:item>
+					<doc:item>
+						<doc:term>left-index-finger</doc:term>
+						<doc:definition>
+							Left index finger
+						</doc:definition>
+					</doc:item>
+					<doc:item>
+						<doc:term>left-middle-finger</doc:term>
+						<doc:definition>
+							Left middle finger
+						</doc:definition>
+					</doc:item>
+					<doc:item>
+						<doc:term>left-ring-finger</doc:term>
+						<doc:definition>
+							Left ring finger
+						</doc:definition>
+					</doc:item>
+					<doc:item>
+						<doc:term>left-little-finger</doc:term>
+						<doc:definition>
+							Left little finger
+						</doc:definition>
+					</doc:item>
+					<doc:item>
+						<doc:term>right-thumb</doc:term>
+						<doc:definition>
+							Right thumb
+						</doc:definition>
+					</doc:item>
+					<doc:item>
+						<doc:term>right-index-finger</doc:term>
+						<doc:definition>
+							Right index finger
+						</doc:definition>
+					</doc:item>
+					<doc:item>
+						<doc:term>right-middle-finger</doc:term>
+						<doc:definition>
+							Right middle finger
+						</doc:definition>
+					</doc:item>
+					<doc:item>
+						<doc:term>right-ring-finger</doc:term>
+						<doc:definition>
+							Right ring finger
+						</doc:definition>
+					</doc:item>
+					<doc:item>
+						<doc:term>right-little-finger</doc:term>
+						<doc:definition>
+							Right little finger
+						</doc:definition>
+					</doc:item>
+					<doc:item>
+						<doc:term>any</doc:term>
+						<doc:definition>
+							Any finger. This is only used for <doc:ref type="method" to="Device.VerifyStart">Device.VerifyStart</doc:ref>
+							(select the first finger with a fingerprint associated, or all the fingerprints available for the user when
+							the device supports it) and <doc:ref type="signal" to="Device::VerifyFingerSelected">Device::VerifyFingerSelected</doc:ref>
+							(any finger with an associated fingerprint can be used).
+						</doc:definition>
+					</doc:item>
+				</doc:list>
+			</doc:para>
+
+			<doc:title id="verify-statuses">
+				Verify Statuses
+			</doc:title>
+			<doc:para>
+				<doc:list>
+					Possible values for the result passed through <doc:ref type="signal" to="Device::VerifyResult">Device::VerifyResult</doc:ref> are:
+					<doc:item>
+						<doc:term>verify-no-match</doc:term>
+						<doc:definition>
+							The verification did not match, <doc:ref type="method" to="Device.VerifyStop">Device.VerifyStop</doc:ref> should now be called.
+						</doc:definition>
+					</doc:item>
+					<doc:item>
+						<doc:term>verify-match</doc:term>
+						<doc:definition>
+							The verification succeeded, <doc:ref type="method" to="Device.VerifyStop">Device.VerifyStop</doc:ref> should now be called.
+						</doc:definition>
+					</doc:item>
+					<doc:item>
+						<doc:term>verify-retry-scan</doc:term>
+						<doc:definition>
+							The user should retry scanning their finger, the verification is still ongoing.
+						</doc:definition>
+					</doc:item>
+					<doc:item>
+						<doc:term>verify-swipe-too-short</doc:term>
+						<doc:definition>
+							The user's swipe was too short. The user should retry scanning their finger, the verification is still ongoing.
+						</doc:definition>
+					</doc:item>
+					<doc:item>
+						<doc:term>verify-finger-not-centered</doc:term>
+						<doc:definition>
+							The user's finger was not centered on the reader. The user should retry scanning their finger, the verification is still ongoing.
+						</doc:definition>
+					</doc:item>
+					<doc:item>
+						<doc:term>verify-remove-and-retry</doc:term>
+						<doc:definition>
+							The user should remove their finger from the reader and retry scanning their finger, the verification is still ongoing.
+						</doc:definition>
+					</doc:item>
+					<doc:item>
+						<doc:term>verify-disconnected</doc:term>
+						<doc:definition>
+							The device was disconnected during the verification, no other actions should be taken, and you shouldn't use the device any more.
+						</doc:definition>
+					</doc:item>
+					<doc:item>
+						<doc:term>verify-unknown-error</doc:term>
+						<doc:definition>
+							An unknown error occurred (usually a driver problem), <doc:ref type="method" to="Device.VerifyStop">Device.VerifyStop</doc:ref> should now be called.
+						</doc:definition>
+					</doc:item>
+				</doc:list>
+			</doc:para>
+
+			<doc:title id="enroll-statuses">
+				Enroll Statuses
+			</doc:title>
+			<doc:para>
+				<doc:list>
+					Possible values for the result passed through <doc:ref type="signal" to="Device::EnrollResult">Device::EnrollResult</doc:ref> are:
+					<doc:item>
+						<doc:term>enroll-completed</doc:term>
+						<doc:definition>
+							The enrollment successfully completed, <doc:ref type="method" to="Device.EnrollStop">Device.EnrollStop</doc:ref> should now be called.
+						</doc:definition>
+					</doc:item>
+					<doc:item>
+						<doc:term>enroll-failed</doc:term>
+						<doc:definition>
+							The enrollment failed, <doc:ref type="method" to="Device.EnrollStop">Device.EnrollStop</doc:ref> should now be called.
+						</doc:definition>
+					</doc:item>
+					<doc:item>
+						<doc:term>enroll-stage-passed</doc:term>
+						<doc:definition>
+							One stage of the enrollment passed, the enrollment is still ongoing.
+						</doc:definition>
+					</doc:item>
+					<doc:item>
+						<doc:term>enroll-retry-scan</doc:term>
+						<doc:definition>
+							The user should retry scanning their finger, the enrollment is still ongoing.
+						</doc:definition>
+					</doc:item>
+					<doc:item>
+						<doc:term>enroll-swipe-too-short</doc:term>
+						<doc:definition>
+							The user's swipe was too short. The user should retry scanning their finger, the enrollment is still ongoing.
+						</doc:definition>
+					</doc:item>
+					<doc:item>
+						<doc:term>enroll-finger-not-centered</doc:term>
+						<doc:definition>
+							The user's finger was not centered on the reader. The user should retry scanning their finger, the enrollment is still ongoing.
+						</doc:definition>
+					</doc:item>
+					<doc:item>
+						<doc:term>enroll-remove-and-retry</doc:term>
+						<doc:definition>
+							The user should remove their finger from the reader and retry scanning their finger, the enrollment is still ongoing.
+						</doc:definition>
+					</doc:item>
+					<doc:item>
+						<doc:term>enroll-data-full</doc:term>
+						<doc:definition>
+							No further prints can be enrolled on this device, <doc:ref type="method" to="Device.EnrollStop">Device.EnrollStop</doc:ref> should now be called.
+
+							<doc:ref type="method" to="DeleteEnrolledFingers2">Delete other prints</doc:ref> from the device first to continue
+							(e.g. from other users). Note that old prints or prints from other operating systems may be deleted automatically
+							to resolve this error without any notification.
+						</doc:definition>
+					</doc:item>
+					<doc:item>
+						<doc:term>enroll-disconnected</doc:term>
+						<doc:definition>
+							The device was disconnected during the enrollment, no other actions should be taken, and you shouldn't use the device any more.
+
+						</doc:definition>
+					</doc:item>
+					<doc:item>
+						<doc:term>enroll-unknown-error</doc:term>
+						<doc:definition>
+							An unknown error occurred (usually a driver problem), <doc:ref type="method" to="Device.EnrollStop">Device.EnrollStop</doc:ref> should now be called.
+
+						</doc:definition>
+					</doc:item>
+				</doc:list>
+			</doc:para>
+		</doc:doc>
+
+		<!-- ************************************************************ -->
+
+		<method name="ListEnrolledFingers">
+			<arg type="s" name="username" direction="in">
+				<doc:doc><doc:summary>The username for whom to list the enrolled fingerprints. See <doc:ref type="description" to="usernames">Usernames</doc:ref>.</doc:summary></doc:doc>
+			</arg>
+			<arg type="as" name="enrolled_fingers" direction="out">
+				<doc:doc><doc:summary>An array of strings representing the enrolled fingerprints. See <doc:ref type="description" to="fingerprint-names">Fingerprint names</doc:ref>.</doc:summary></doc:doc>
+			</arg>
+			<annotation name="org.freedesktop.DBus.GLib.Async" value="" />
+
+			<doc:doc>
+				<doc:description>
+					<doc:para>
+						List all the enrolled fingerprints for the chosen user.
+					</doc:para>
+				</doc:description>
+
+				<doc:errors>
+					<doc:error name="&ERROR_PERMISSION_DENIED;">if the caller lacks the appropriate PolicyKit authorization</doc:error>
+					<doc:error name="&ERROR_NO_ENROLLED_PRINTS;">if the chosen user doesn't have any fingerprints enrolled</doc:error>
+				</doc:errors>
+			</doc:doc>
+		</method>
+
+		<!-- ************************************************************ -->
+
+		<method name="DeleteEnrolledFingers">
+			<arg type="s" name="username" direction="in">
+				<doc:doc><doc:summary>The username for whom to delete the enrolled fingerprints. See <doc:ref type="description" to="usernames">Usernames</doc:ref>.</doc:summary></doc:doc>
+			</arg>
+			<annotation name="org.freedesktop.DBus.GLib.Async" value="" />
+
+			<doc:doc>
+				<doc:description>
+					<doc:para>
+						Delete all the enrolled fingerprints for the chosen user.
+					</doc:para>
+					<doc:para>
+						This call only exists for compatibility reasons, you should instead claim the device using
+						<doc:ref type="method" to="Device.Claim">Device.Claim</doc:ref> and then call
+						<doc:ref type="method" to="DeleteEnrolledFingers2">DeleteEnrolledFingers2</doc:ref>.
+					</doc:para>
+				</doc:description>
+
+				<doc:errors>
+					<doc:error name="&ERROR_PERMISSION_DENIED;">if the caller lacks the appropriate PolicyKit authorization</doc:error>
+				</doc:errors>
+			</doc:doc>
+		</method>
+
+		<!-- ************************************************************ -->
+
+		<method name="DeleteEnrolledFingers2">
+			<annotation name="org.freedesktop.DBus.GLib.Async" value="" />
+
+			<doc:doc>
+				<doc:description>
+					<doc:para>
+						Delete all the enrolled fingerprints for the user currently claiming the device with <doc:ref type="method" to="Device.Claim">Device.Claim</doc:ref>.
+					</doc:para>
+				</doc:description>
+
+				<doc:errors>
+					<doc:error name="&ERROR_PERMISSION_DENIED;">if the caller lacks the appropriate PolicyKit authorization</doc:error>
+				</doc:errors>
+			</doc:doc>
+		</method>
+
+		<!-- ************************************************************ -->
+
+		<method name="Claim">
+			<arg type="s" name="username" direction="in">
+				<doc:doc><doc:summary>The username for whom to claim the device. See <doc:ref type="description" to="usernames">Usernames</doc:ref>.</doc:summary></doc:doc>
+			</arg>
+			<annotation name="org.freedesktop.DBus.GLib.Async" value="" />
+
+			<doc:doc>
+				<doc:description>
+					<doc:para>
+						Claim the device for the chosen user.
+					</doc:para>
+				</doc:description>
+
+				<doc:errors>
+					<doc:error name="&ERROR_PERMISSION_DENIED;">if the caller lacks the appropriate PolicyKit authorization</doc:error>
+					<doc:error name="&ERROR_ALREADY_IN_USE;">if the device is already claimed</doc:error>
+					<doc:error name="&ERROR_INTERNAL;">if the device couldn't be claimed</doc:error>
+				</doc:errors>
+			</doc:doc>
+		</method>
+
+		<!-- ************************************************************ -->
+
+		<method name="Release">
+			<annotation name="org.freedesktop.DBus.GLib.Async" value="" />
+
+			<doc:doc>
+				<doc:description>
+					<doc:para>
+						Release a device claimed with <doc:ref type="method" to="Device.Claim">Device.Claim</doc:ref>.
+					</doc:para>
+				</doc:description>
+
+				<doc:errors>
+					<doc:error name="&ERROR_PERMISSION_DENIED;">if the caller lacks the appropriate PolicyKit authorization</doc:error>
+					<doc:error name="&ERROR_CLAIM_DEVICE;">if the device was not claimed</doc:error>
+				</doc:errors>
+			</doc:doc>
+		</method>
+
+		<!-- ************************************************************ -->
+
+		<method name="VerifyStart">
+			<arg type="s" name="finger_name" direction="in">
+				<doc:doc><doc:summary>A string representing the finger to verify. See <doc:ref type="description" to="fingerprint-names">Fingerprint names</doc:ref>.</doc:summary></doc:doc>
+			</arg>
+			<annotation name="org.freedesktop.DBus.GLib.Async" value="" />
+
+			<doc:doc>
+				<doc:description>
+					<doc:para>
+						Check the chosen finger against a saved fingerprint. You need to have claimed the device using
+						<doc:ref type="method" to="Device.Claim">Device.Claim</doc:ref>. The finger selected is sent to the front-end
+						using <doc:ref type="signal" to="Device::VerifyFingerSelected">Device::VerifyFingerSelected</doc:ref> and
+						verification status through <doc:ref type="signal" to="Device::VerifyStatus">Device::VerifyStatus</doc:ref>.
+					</doc:para>
+				</doc:description>
+
+				<doc:errors>
+					<doc:error name="&ERROR_PERMISSION_DENIED;">if the caller lacks the appropriate PolicyKit authorization</doc:error>
+					<doc:error name="&ERROR_CLAIM_DEVICE;">if the device was not claimed</doc:error>
+					<doc:error name="&ERROR_ALREADY_IN_USE;">if the device was already being used</doc:error>
+					<doc:error name="&ERROR_NO_ENROLLED_PRINTS;">if there are no enrolled prints for the chosen user</doc:error>
+					<doc:error name="&ERROR_INTERNAL;">if there was an internal error</doc:error>
+				</doc:errors>
+			</doc:doc>
+		</method>
+
+		<!-- ************************************************************ -->
+
+		<method name="VerifyStop">
+			<annotation name="org.freedesktop.DBus.GLib.Async" value="" />
+
+			<doc:doc>
+				<doc:description>
+					<doc:para>
+						Stop an on-going fingerprint verification started with <doc:ref type="method" to="Device.VerifyStart">Device.VerifyStart</doc:ref>.
+					</doc:para>
+				</doc:description>
+
+				<doc:errors>
+					<doc:error name="&ERROR_PERMISSION_DENIED;">if the caller lacks the appropriate PolicyKit authorization</doc:error>
+					<doc:error name="&ERROR_CLAIM_DEVICE;">if the device was not claimed</doc:error>
+					<doc:error name="&ERROR_NO_ACTION_IN_PROGRESS;">if there was no ongoing verification</doc:error>
+					<doc:error name="&ERROR_INTERNAL;">if there was an internal error</doc:error>
+				</doc:errors>
+			</doc:doc>
+		</method>
+
+		<!-- ************************************************************ -->
+
+		<signal name="VerifyFingerSelected">
+			<arg type="s" name="finger_name">
+				<doc:doc>
+					<doc:summary>
+						<doc:para>
+							A string representing the finger select to be verified.
+						</doc:para>
+					</doc:summary>
+				</doc:doc>
+			</arg>
+			<doc:doc>
+				<doc:seealso>
+					<doc:ref type="description" to="fingerprint-names">Fingerprint names</doc:ref>.
+				</doc:seealso>
+			</doc:doc>
+		</signal>
+
+		<!-- ************************************************************ -->
+
+		<signal name="VerifyStatus">
+			<arg type="s" name="result">
+				<doc:doc>
+					<doc:summary>
+						A string representing the status of the verification.
+					</doc:summary>
+				</doc:doc>
+			</arg>
+
+			<arg type="b" name="done">
+				<doc:doc>
+					<doc:summary>
+						Whether the verification finished and can be stopped.
+					</doc:summary>
+				</doc:doc>
+			</arg>
+
+			<doc:doc>
+				<doc:seealso>
+					<doc:ref type="description" to="verify-statuses">Verify Statuses</doc:ref> and <doc:ref type="method" to="Device.VerifyStop">Device.VerifyStop</doc:ref>.
+				</doc:seealso>
+			</doc:doc>
+		</signal>
+
+		<!-- ************************************************************ -->
+
+		<method name="EnrollStart">
+			<arg type="s" name="finger_name" direction="in">
+				<doc:doc><doc:summary>A string representing the finger to enroll. See
+				<doc:ref type="description" to="fingerprint-names">Fingerprint names</doc:ref>.
+				Note that "any" is not a valid finger name for this method.</doc:summary></doc:doc>
+			</arg>
+			<annotation name="org.freedesktop.DBus.GLib.Async" value="" />
+
+			<doc:doc>
+				<doc:description>
+					<doc:para>
+						Start enrollment for the selected finger. You need to have claimed the device using
+						<doc:ref type="method" to="Device.Claim">Device.Claim</doc:ref> before calling
+						this method. Enrollment status is sent through <doc:ref type="signal" to="Device::EnrollStatus">Device::EnrollStatus</doc:ref>.
+					</doc:para>
+				</doc:description>
+
+				<doc:errors>
+					<doc:error name="&ERROR_PERMISSION_DENIED;">if the caller lacks the appropriate PolicyKit authorization</doc:error>
+					<doc:error name="&ERROR_CLAIM_DEVICE;">if the device was not claimed</doc:error>
+					<doc:error name="&ERROR_ALREADY_IN_USE;">if the device was already being used</doc:error>
+					<doc:error name="&ERROR_INVALID_FINGERNAME;">if the finger name passed is invalid</doc:error>
+					<doc:error name="&ERROR_INTERNAL;">if there was an internal error</doc:error>
+				</doc:errors>
+
+			</doc:doc>
+		</method>
+
+		<!-- ************************************************************ -->
+
+		<method name="EnrollStop">
+			<annotation name="org.freedesktop.DBus.GLib.Async" value="" />
+
+			<doc:doc>
+				<doc:description>
+					<doc:para>
+						Stop an on-going fingerprint enrollment started with <doc:ref type="method" to="Device.EnrollStart">Device.EnrollStart</doc:ref>.
+					</doc:para>
+				</doc:description>
+
+				<doc:errors>
+					<doc:error name="&ERROR_PERMISSION_DENIED;">if the caller lacks the appropriate PolicyKit authorization</doc:error>
+					<doc:error name="&ERROR_CLAIM_DEVICE;">if the device was not claimed</doc:error>
+					<doc:error name="&ERROR_NO_ACTION_IN_PROGRESS;">if there was no ongoing verification</doc:error>
+					<doc:error name="&ERROR_INTERNAL;">if there was an internal error</doc:error>
+				</doc:errors>
+			</doc:doc>
+		</method>
+
+		<!-- ************************************************************ -->
+
+		<signal name="EnrollStatus">
+			<arg type="s" name="result">
+				<doc:doc>
+					<doc:summary>
+						A string representing the status of the enrollment.
+					</doc:summary>
+				</doc:doc>
+			</arg>
+
+			<arg type="b" name="done">
+				<doc:doc>
+					<doc:summary>
+						Whether the enrollment finished and can be stopped.
+					</doc:summary>
+				</doc:doc>
+			</arg>
+
+			<doc:doc>
+				<doc:seealso>
+					<doc:ref type="description" to="enroll-statuses">Enrollment Statuses</doc:ref> and <doc:ref type="method" to="Device.EnrollStop">Device.EnrollStop</doc:ref>.
+				</doc:seealso>
+			</doc:doc>
+		</signal>
+
+		<!-- ************************************************************ -->
+
+		<property name="name" type="s" access="read">
+			<doc:doc>
+				<doc:description>
+					<doc:para>
+						The product name of the device.
+					</doc:para>
+				</doc:description>
+			</doc:doc>
+		</property>
+
+		<!-- ************************************************************ -->
+
+		<property name="num_enroll_stages" type="i" access="read">
+			<doc:doc>
+				<doc:description>
+					<doc:para>
+						The number of enrollment stages for the device. This is only available when the device has been claimed, otherwise it will be undefined (-1).
+					</doc:para>
+					<doc:seealso>
+						<doc:ref type="method" to="Device.Claim">Device.Claim</doc:ref> and <doc:ref type="method" to="Device.EnrollStart">Device.EnrollStart</doc:ref>.
+					</doc:seealso>
+				</doc:description>
+			</doc:doc>
+		</property>
+
+		<!-- ************************************************************ -->
+
+		<property name="scan_type" type="s" access="read">
+			<doc:doc>
+				<doc:description>
+					<doc:para>
+						The scan type of the device, either "press" if you place your finger on the device, or "swipe" if you have to swipe your finger.
+					</doc:para>
+				</doc:description>
+			</doc:doc>
+		</property>
+
+	</interface>
+</node>
+
diff -Npur plasma-desktop-5.20.5.orig/kcms/users/src/net.reactivated.Fprint.Manager.xml plasma-desktop-5.20.5/kcms/users/src/net.reactivated.Fprint.Manager.xml
--- plasma-desktop-5.20.5.orig/kcms/users/src/net.reactivated.Fprint.Manager.xml	1970-01-01 01:00:00.000000000 +0100
+++ plasma-desktop-5.20.5/kcms/users/src/net.reactivated.Fprint.Manager.xml	2021-01-23 14:04:18.470688454 +0100
@@ -0,0 +1,50 @@
+<!DOCTYPE node PUBLIC
+"-//freedesktop//DTD D-BUS Object Introspection 1.0//EN"
+"http://www.freedesktop.org/standards/dbus/1.0/introspect.dtd" [
+<!ENTITY ERROR_NO_SUCH_DEVICE "net.reactivated.Fprint.Error.NoSuchDevice">
+]>
+<node name="/" xmlns:doc="http://www.freedesktop.org/dbus/1.0/doc.dtd">
+	<interface name="net.reactivated.Fprint.Manager">
+		<annotation name="org.freedesktop.DBus.GLib.CSymbol"
+			value="fprint_manager" />
+
+		<!-- ************************************************************ -->
+
+		<method name="GetDevices">
+			<arg type="ao" name="devices" direction="out">
+				<doc:doc><doc:summary>An array of object paths for devices.</doc:summary></doc:doc>
+			</arg>
+
+			<doc:doc>
+				<doc:description>
+					<doc:para>
+						Enumerate all the fingerprint readers attached to the system. If there are
+						no devices available, an empty array is returned.
+					</doc:para>
+				</doc:description>
+			</doc:doc>
+		</method>
+
+		<!-- ************************************************************ -->
+
+		<method name="GetDefaultDevice">
+			<arg type="o" name="device" direction="out">
+				<doc:doc><doc:summary>The object path for the default device.</doc:summary></doc:doc>
+			</arg>
+
+			<doc:doc>
+				<doc:description>
+					<doc:para>
+						Returns the default fingerprint reader device.
+					</doc:para>
+				</doc:description>
+
+				<doc:errors>
+					<doc:error name="&ERROR_NO_SUCH_DEVICE;">if the device does not exist</doc:error>
+				</doc:errors>
+			</doc:doc>
+		</method>
+
+	</interface>
+</node>
+
