diff -Npur linux-4.17.orig/include/linux/pm_qos.h linux-4.17/include/linux/pm_qos.h
--- linux-4.17.orig/include/linux/pm_qos.h	2018-06-03 23:15:21.000000000 +0200
+++ linux-4.17/include/linux/pm_qos.h	2018-07-28 18:16:23.346207389 +0200
@@ -177,8 +174,7 @@ 
 static inline s32 dev_pm_qos_raw_read_value(struct device *dev)
 {
 	return IS_ERR_OR_NULL(dev->power.qos) ?
-		PM_QOS_RESUME_LATENCY_NO_CONSTRAINT :
-		pm_qos_read_value(&dev->power.qos->resume_latency);
+		0 : pm_qos_read_value(&dev->power.qos->resume_latency);
 }
 #else
 static inline enum pm_qos_flags_status __dev_pm_qos_flags(struct device *dev,
