--- b/drivers/gpu/drm/i915/gt/intel_ring.c
+++ a/drivers/gpu/drm/i915/gt/intel_ring.c
@@ -49,7 +49,7 @@
 	if (unlikely(ret))
 		goto err_unpin;
 
+	if (i915_vma_is_map_and_fenceable(vma))
-	if (i915_vma_is_map_and_fenceable(vma) && !HAS_LLC(vma->vm->i915))
 		addr = (void __force *)i915_vma_pin_iomap(vma);
 	else
 		addr = i915_gem_object_pin_map(vma->obj,
@@ -91,7 +91,7 @@
 		return;
 
 	i915_vma_unset_ggtt_write(vma);
+	if (i915_vma_is_map_and_fenceable(vma))
-	if (i915_vma_is_map_and_fenceable(vma) && !HAS_LLC(vma->vm->i915))
 		i915_vma_unpin_iomap(vma);
 	else
 		i915_gem_object_unpin_map(vma->obj);
