--- ladspa-sdk.orig/src/Makefile	2024-09-21 11:40:18.143358953 +0700
+++ ladspa-sdk/src/Makefile		2024-09-21 11:42:12.730026698 +0700
@@ -4,9 +4,9 @@
 #
 # Change these if you want to install somewhere else.
 
-INSTALL_PLUGINS_DIR	=	/usr/lib/ladspa/
-INSTALL_INCLUDE_DIR	=	/usr/include/
-INSTALL_BINARY_DIR	=	/usr/bin/
+INSTALL_PLUGINS_DIR	=	$(DESTDIR)/usr/lib/ladspa/
+INSTALL_INCLUDE_DIR	=	$(DESTDIR)/usr/include/
+INSTALL_BINARY_DIR	=	$(DESTDIR)/usr/bin/
 
 ###############################################################################
 #
